woocommerce/woocommerce-android
# [Customer Search] Pass selected customer to the customer details screen

Closes: #9306


### Description

The PR adds the passing of the selected customer to the next screen. It uses the old to do so because it seems that changing that maybe take quite some time. If we go further with the project, this may be m2 thing

Notice that there is no "partial" loading state yet (when we fetch a customer)

### Testing instructions


* Start order creation
* Add customer details
* Search/select any customer
* Notice that the data is passed to the next screen


* Test the same with feature flag OFF (release).
* Notice that nothing's changed to the old flow


### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/4923871/0e209a4c-6d17-4e6d-bb97-0d09542d3b61
